csrc/device_lower/analysis/device_version.h

```diff
@@ -0,0 +1,59 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <c10/macros/Export.h>
+#include <exceptions.h>
+
+#include <compute_at_map.h>
+#include <dispatch.h>
+#include <ir/all_nodes.h>
+#include <kernel_ir.h>
+
+#include <vector>
+
+namespace nvfuser {
+
+class LoopIndexing;
+class ComputeAtMap;
+
+//! Traverses a Fusion to find the minimum supported CUDA compute capability
+//! that will be able to run the generated kernel.
+class MinimumDeviceVersion : IterVisitor {
```

Comment:
```suggestion
class MinimumDeviceVersion : private IterVisitor {
```

I rarely used private inheritance so I forgot about the default :) 

Nit: I'm not a big fan of using inheritance when it's not an "is-a" relation. I'd probably do composition instead -- adding a public IterVisitor as a member.